src/Unosquare.Swan.Lite/Formatters/Json.Converter.cs

```diff
@@ -0,0 +1,369 @@
+﻿namespace Unosquare.Swan.Formatters
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using System.Text;
+    using Attributes;
+
+    /// <summary>
+    /// A very simple, light-weight JSON library written by Mario
+    /// to teach Geo how things are done
+    /// 
+    /// This is an useful helper for small tasks but it doesn't represent a full-featured
+    /// serializer such as the beloved Json.NET
+    /// </summary>
+    public static partial class Json
+    {
+        private class Converter
+        {
+            private static readonly Dictionary<MemberInfo, string> MemberInfoNameCache =
+                new Dictionary<MemberInfo, string>();
+            private static readonly Dictionary<Type, Type> ListAddMethodCache = new Dictionary<Type, Type>();
+
+            private readonly object _target;
+            private readonly Type _targetType;
+            private readonly bool _includeNonPublic;
+
+            private Converter(
+                object source,
+                Type targetType,
+                ref object targetInstance,
+                bool includeNonPublic)
+            {
+                _targetType = targetType;
+
+                if (source == null)
+                {
+                    _target = _targetType.GetDefault();
+                    return;
+                }
+
+                _includeNonPublic = includeNonPublic;
+                var sourceType = source.GetType();
+
+                if (targetInstance != null) _targetType = targetInstance.GetType();
+                if (_targetType == null || _targetType == typeof(object)) _targetType = sourceType;
+                if (sourceType == _targetType)
+                {
+                    _target = source;
+                    return;
+                }
+
+                if (targetInstance == null)
+                {
+                    // Try to create a default instance
+                    try
+                    {
+                        source.CreateTarget(_targetType, _includeNonPublic, ref _target);
+                    }
+                    catch
+                    {
+                        _target = _targetType.GetDefault();
+                        return;
+                    }
+                }
+                else
+                {
+                    _target = targetInstance;
+                }
```

Comment:
return is missing?

Reply:
Nop, the target is required in the following lines.